src/instrument.h

```diff
@@ -7,16 +7,26 @@
 #define _INSTRUMENT_H
 
 #include <jvmti.h>
+#include <string>
+#include <unordered_map>
+#include <vector>
 #include "arch.h"
 #include "engine.h"
 
+typedef std::string ClassName;
+typedef std::string MethodName;
+typedef std::string MethodSignature;
+typedef long Latency;
+typedef std::unordered_map<MethodName, std::unordered_map<MethodSignature, Latency>> MethodTargets;
+typedef std::unordered_map<ClassName, MethodTargets> Targets;
```

Comment:
No to 3-level maps.
I can understand why 2 levels may be necessary (because you have to make an early decision based on a class name), but 3 is overkill.

Reply:
I think things are a bit uglier this way: 6ff4fc6fb0a6e1f03349b02179941869460044ba

Am I missing some obvious simplification?